internal/pkg/cli/job_package.go

```diff
@@ -0,0 +1,180 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/term/command"
+	"github.com/aws/copilot-cli/internal/pkg/term/prompt"
+	"github.com/aws/copilot-cli/internal/pkg/term/selector"
+	"github.com/aws/copilot-cli/internal/pkg/workspace"
+)
+
+const (
+	jobPackageJobNamePrompt = "Which job would you like to generate a CloudFormation template for?"
+	jobPackageEnvNamePrompt = "Which environment would you like to package this stack for?"
+)
+
+type packageJobVars struct {
+	name      string
+	envName   string
+	appName   string
+	tag       string
+	outputDir string
+}
+
+type packageJobOpts struct {
+	packageJobVars
+
+	// Interfaces to interact with dependencies.
+	ws     wsJobDirReader
+	store  store
+	runner runner
+	sel    wsSelector
+	prompt prompter
+}
+
+func newPackageJobOpts(vars packageJobVars) (*packageJobOpts, error) {
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, fmt.Errorf("new workspace: %w", err)
+	}
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("connect to config store: %w", err)
+	}
+	prompter := prompt.New()
+	opts := &packageJobOpts{
+		packageJobVars: vars,
+		ws:             ws,
+		store:          store,
+		runner:         command.New(),
+		sel:            selector.NewWorkspaceSelect(prompter, store, ws),
+		prompt:         prompter,
+	}
+	return opts, nil
+}
+
+// Validate returns an error if the values provided by the user are invalid.
+func (o *packageJobOpts) Validate() error {
+	if o.appName == "" {
+		return errNoAppInWorkspace
+	}
+	if o.name != "" {
+		names, err := o.ws.JobNames()
+		if err != nil {
+			return fmt.Errorf("list jobs in the workspace: %w", err)
+		}
+		if !contains(o.name, names) {
+			return fmt.Errorf("job '%s' does not exist in the workspace", o.name)
+		}
+	}
+	if o.envName != "" {
+		if _, err := o.store.GetEnvironment(o.appName, o.envName); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Ask prompts the user for any missing required fields.
+func (o *packageJobOpts) Ask() error {
+	if err := o.askAppName(); err != nil {
+		return err
+	}
+	if err := o.askEnvName(); err != nil {
+		return err
+	}
+	return o.askTag()
+}
+
+// Execute prints the CloudFormation template of the application for the environment.
+func (o *packageJobOpts) Execute() error {
+	return nil
+}
+
+func (o *packageJobOpts) askAppName() error {
```

Comment:
Should this method be renamed to `askJobName`?

Reply:
I think this is a holdover from The Great App-to-Service Rename of 2020! https://github.com/aws/copilot-cli/pull/919/files. Fixing here and in `svc_package`.